horizomer/utils/tree.py

```diff
@@ -44,6 +44,31 @@ def compare_topology(tree1, tree2):
     return n2p1 == n2p2
 
 
+def intersect_trees(tree1, tree2):
+    """Shrink two trees to contain only overlapping taxa.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree to intersect
+    tree2 : skbio.TreeNode
+        second tree to intersect
+
+    Returns
+    -------
+    tuple of two TreeNodes
+        resulting trees containing only overlapping taxa
+    """
+    taxa1 = set([tip.name for tip in tree1.tips()])
```

Comment:
what if tip.name is None?

Reply:
A tip must have a name.

Reply:
nope:
```
t = TreeNode.read(['(a,);'])
for tip in t.tips():
    print(tip.name)
```

Reply:
Good experiment!

I think if we test tip name uniqueness explicitly, this `None` check can be omitted.

Reply:
Addressed this comment.